content/engineering-practices-sdlc-improvements.md

```diff
@@ -0,0 +1,82 @@
+Title: Exploring the future Software Development Lifecycle (SDLC) flow
+Date: 2024-06-07
+Category: Posts 
+Tags: azure-devops, engineering, automation
+Slug: engineering-practices-sdlc-improvements
+Author: Willy-Peter Schaub
+Summary: Looking for Insights and Feedback from other pioneers of continuous SDLC improvements.
+
+We want to "_Unite people, process, and products to empower and enable us to deliver continuous value to delighted end-users_", following [Donovan's definition of DevOps](https://www.donovanbrown.com/post/what-is-devops), and our aim is to "_Create an SDLC with consistent, standardized, secure, simple, and cost-effective processes, practices, and products_". We always try to improve our people (80%), processes (15%), and products (5%). In this post I will highlight one topic we are investigating and I hope you will join us and share your views on whether we are on track, or need some adjustment – **SDLC Flow**.
+
+---
+
+# SDLC / FLOW / TODAY
+
+> ![Today's SDLC](../images/engineering-practices-sdlc-improvements-1.png) 
+
+As illustrated, our current software development lifecycle (SDLC) flow involves four main environments – sometimes fewer, sometimes more, but this is the usual pattern. Development, unit testing, and verification are performed in the **development** environment, integration testing, performance testing, and user acceptance testing are conducted in the **system test** environment, additional user acceptance testing and ad-hoc bug analysis are carried out in **Staging**, and we are all familiar with what happens in **Production**.  
+
+### Advantages
+
+- Clear separation of duties.
+- Capacity to work on and deliver a lot of value (super tankers).
+
+### Disadvantages
+
+- Multiple environments that need careful operational attention.
+- Multiple environments that produce carbon emissions.
+- Multiple environments that can diverge from each other in terms of data, setup, 3rd party software, for example .NET, and our solution artifacts.
+- Multiple environments that increase **complexity** and **total cost of ownership**.
+
+---
+
+# SDLC / FLOW / FUTURE
+
+> ![SDLC of the future](../images/engineering-practices-sdlc-improvements-2.png) 
+
+We want to move as much of our quality and security scans to the left as we can, adopt a mindset of continuous delivery and release on demand, and decrease the number of environments. We want to gradually show features in a new (green) version, while keeping the live production (blue) version. That way we can switch from green to blue and switch back if needed.  
```

Comment:
... scan to the left as we can, ... 
should be:
...scan to the left as far as we can, ...

Reply:
Fixed, thanks for catching that.